README.md

```diff
@@ -34,7 +34,70 @@ in the contribution by Working Group III to the IPCC's Sixth Assessment Report (
 Please refer to the Horizon 2020 project [openENTRANCE](https://github.com/openENTRANCE/openentrance#data-format-structure)
 for more information about the format and its usage in that project.
 
-## Usage for OSeMOSYS
+## Writing a configuration file
+
+Write a configuration file in YAML format. A simple configuration file with one result variable looks like this:
+
+    model: OSeMBE v1.0.0
+    scenario: DIAG-C400-lin-ResidualFossil
+    region: ['Austria']
+    results:
+    - iamc_variable: 'Carbon Capture|Biomass'
+    tech_emi: ['(?=^.{2}(BM))^.{4}(CS)']
+    emissions: [CO2]
+    unit: kt CO2/yr
+    transform: abs
+    osemosys_param: AnnualTechnologyEmissions
+
+The first section of the configuration file with the keys `model`, `scenario`, and `region` are used to define the metadata for
+the IAMC template.
+
+The second section `results:` is where you describe each of the IAMC variables and provide instructions to osemosys2iamc on how
+to compute the values.
+
+`iamc_variable` - this should match one of the IAMC variable names
+`unit` - provide the units of the OSeMOSYS results
+`transform` - only `abs` is currently available. This returns the absolute value of the results
+`osemosys_param` - provide the name of the result file from which the script should extract the result data
+
+One or more of the following filter keys. These filter the
+results by one or more columns found in OSeMOSYS results.
+Following the fitering, the remaining columns except region
+ and year are discarded and rows are summed.
+
+`tech_emi` - filter the results by TECHNOLOGY and EMISSION columns using the provide regular expression and an `emissions` entry
+`emissions` - a list of emissions to filter the results by the EMISSION column
+`fuel` - filter by the FUEL column
+`capacity` - filter the TECHNOLOGY column
+`primary_technology` - filter the TECHNOLOGY column (can be replaced by `capacity` key)
+`excluded_prod_tech` - filter the TECHNOLOGY column (can be replaced by `capacity` key)
+`el_prod_technology` - filter the TECHNOLOGY column (can be replaced by `capacity` key)
+`demand` - filters by the FUEL column (final energy)
+
+The value for each of these keys is a list of regular expressions. These regular expressions are used to filter the rows of data in the chosen column to those that match the
+regular expression.
+
+Writing regular expressions can be tricky, but there are [useful tools](https://regexr.com/) to help.
+Below we provide some examples:
+
+`^.{2}(WI)` match rows with exactly two characters followed by `WI`
```

Comment:
I think this formulation might be misunderstood, even though it is correct.
For example when filtering a `TECHNOLOGY` column with a technology called `ATWIONPH1` the regex `^.{2}(WI)` will select the rows with `ATWIONPH1`, because the first four letters match the pattern.
Perhaps we could write something like: `^.{2}(WI)` matches rows which contain a string of four characters or more that starts with two non specified charactes followed by `WI`.